### PR TITLE
gh17743 new GH action and script to create a...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

new GH action and script to create a branch once every 7.yy.0 release (to be called by che-release script)

Change-Id: Ie206b320763d98444a53003c24f14fb76831b5e0
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17743

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
@nickboldt this script will only be used during major version releases? You don't need to "mark" minor versions, right?

**Comment by nickboldt** [comment]
point is to only create a 7.yy.x branch, so for minor .z releases the branch should already exist.

**Comment by mkuznyetsov** [comment]
right, but say, for version 7.27.0 we make 7.27.x branch here, then for 7.21.0 we don't do anything, even though there might be changes in main branch of this project?

**Comment by nickboldt** [comment]
Yes, that's the whole point. Today, every time there's a CRW release, I manually cut a fresh .x branch to use for the ER and RC builds (7.23.x, 7.24.x). 

In future, I'd like to not have to do that... like the other 10 GH projects on which CRW depends, it would be great if there was already a 7.26.x branch when I need one for CRW 2.7.

**Comment by mkuznyetsov** [comment]
OK, I see, I was just thinking if you need something to somehow distinguish between minor versions, while you only have just bare .x branches, which what looks like to be enough for you

File: .github/workflows/make-branch.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Create branch
+on:
+  # to be run once every 3 weeks for a 7.yy.0 release, so we can get a stable 7.yy.x branch for use by downstream consumers
+  workflow_dispatch:
+    inputs:
```
Comment (mkuznyetsov) [inline .github/workflows/make-branch.yaml:5]: Might as well add --force parameter (as well as in other PR)

**Review by mkuznyetsov — APPROVED** [review]

**Pull request merged**